### ISSUE TITLE
Mess in response codes in synchronous client

### ISSUE THREAD

**Issue by AdrianMoranMontes** [eclipse-paho/paho.mqtt.c#834]
**Describe the bug**
Mess in response codes in synchronous client

**To Reproduce**
Do not generate errors in testinig suite, but yes it does in 3rd party platforms.

**Expected behavior**
Codes assigned to `enum MQTTReasonCodes` variable should be in the range of valid values.

**Additional context**
There are lot of places in `MQTTClient.c` where `MQTTResponse` type is used. MQTTResponse struct has a field named `reasonCode`, which is of type `enum MQTTReasonCodes`. Such enum defines a set of values (all positive).

However, in lot of places of `MQTTClient.c`, negative values are assigned to such `reasonCode` field. Some examples:

- Function `MQTTClient_connectURIVersion`.
- Function `MQTTClient_connectURI`.
- Function `MQTTClient_connectAll`.
- Function `MQTTClient_subscribeMany5`.
- ...

There are platforms where it is not possible to assign a value to a enum outside of the valid range.

It a mess for me (I do not know which is the desired behavior) so I report the issue without a PR with the solution.

Regards,
Adrian

**Comment by icraggs** [comment]
From my copy of the C Programming Language 2nd Edition (ANSI C):

_Although variables of enum types may be declared, compilers need not check that what you store in such a variable is a valid value for the enumeration._

This isn't C++, it's C, for better or worse.  I've always viewed enums as guidelines or extra help for integers.  I'd be interested in what any of the latest C standards say - I don't have any to hand at the moment.

**Comment by AdrianMoranMontes** [comment]
Sure Ian, you are right.

If a compile my program with a C++ compile (g++) it does not complain about that, and it works perfectly. I found the problem compiling muy program with a little bit "special" compiler (GreenHills compiler for Integrity), and **in runtime** the program stops, complaining about assigning a value to enum that is not in the valid range.

I thought it was not intentionally developed this way, so I opened the issue. Considering you were already aware of this, I close the issue to avoid more disturbing.

Best regards,
Adrian
